src/driver-roccat-kone-aimo.c

```diff
@@ -0,0 +1,876 @@
+
+struct roccat_color {
+	uint8_t intensity;
+	uint8_t red;
+	uint8_t green;
+	uint8_t blue;
+	uint8_t padding;
+} __attribute__((packed));
+
+struct roccat_settings_report {
+	uint8_t reportID; // 6
+	uint8_t reportDataLength; // 126 for settings.
+	uint8_t profileID; // 5 Profiles, 0-4
+	uint8_t sensitivity; /* 0x06 means 0 because its -5 to 5 in the UI, so 1-11*/
+	uint8_t x_y_linked; // I assume this is for being able to set X and Y dpi separately, but its not a feature in ROCCAT Swarm, default `1F`
+	uint8_t current_dpi; // 0-4 for the 5 resolutions
+	uint16_t xres[5]; // 5 resolutions saved to switch between. value * 50 = DPI
+	uint16_t yres[5]; // 5 resolutions saved to switch between. value * 50 = DPI
+	uint8_t report_rate; // 0 = 125 hz, 1 = 250hz, 2 = 500hz, 3 = 1000hz
+	uint8_t angle_snapping; // 0 off, 1 on
+	uint8_t padding1;
+	struct roccat_color unknown_color1; // 08 FF 07 00 not sure, looks like some kind of RED color at low intensity?
+	uint8_t effect_type; /* 01 = fixed, 06 = snake, 0a = wave, 08 = heartbeat2.0, 07 = breathing2.0, 09 = AIMO */
+	uint8_t lightfx_speed; /* 1-11 */
+	struct roccat_color brightness_color; // intensity is global brightness, color is unused but `1D 13 FF` default.
+	struct roccat_color unknown_color3; // FF 59 FF 00, looks like some kind of default gradient, might be AIMO gradient?
+	struct roccat_color unknown_color4; // FF FD FD 00
+	struct roccat_color unknown_color5; // FF F4 64 00
+	struct roccat_color unknown_color6; // FF F4 00 00
+	uint8_t unknown1; // FF default
+	struct roccat_color led_scrollwheel_color; // only solid
+	struct roccat_color led_leftstrip_color_1; // 4 color gradient. Solid is just all same color
+	struct roccat_color led_leftstrip_color_2;
+	struct roccat_color led_leftstrip_color_3;
+	struct roccat_color led_leftstrip_color_4;
+    struct roccat_color led_rightstrip_color_1; // 4 color gradient. Solid is just all same color
+	struct roccat_color led_rightstrip_color_2;
+	struct roccat_color led_rightstrip_color_3;
+	struct roccat_color led_rightstrip_color_4;
+	struct roccat_color led_leftblob_color; // only solid
+	struct roccat_color led_rightblob_color; // only solid
+	uint8_t custom_or_theme; // 00-09 for custom based on theme, theme is 80-89
+	uint8_t unknown2; // 01 default
+	uint8_t padding_end[6]; // All 0s.
+	uint16_t checksum;
+} __attribute__((packed));
+
+struct roccat_macro {
+	uint8_t reportID;
+	uint8_t twentytwo;
+	uint8_t height;
+	uint8_t profile;
+	uint8_t button_index;
+	uint8_t active;
+	uint8_t padding[24];
+	char group[24];
+	char name[24];
+	uint16_t length;
+	struct {
+		uint8_t keycode;
+		uint8_t flag;
+		uint16_t time;
+	} keys[ROCCAT_MAX_MACRO_LENGTH];
+	uint16_t checksum;
+} __attribute__((packed));
+
+struct roccat_data {
+	uint8_t profiles[(ROCCAT_PROFILE_MAX )][ROCCAT_REPORT_SIZE_PROFILE];
+	struct roccat_settings_report settings[(ROCCAT_PROFILE_MAX )];
+	struct roccat_macro macros[(ROCCAT_PROFILE_MAX )][(ROCCAT_BUTTON_MAX)];
```

Comment:
This has an extra space after `ROCCAT_PROFILE_MAX` (`(ROCCAT_PROFILE_MAX )`).

Reply:
All set